backend/src/main/java/kr/momo/domain/attendee/AttendeeGroup.java

```diff
@@ -0,0 +1,77 @@
+package kr.momo.domain.attendee;
+
+import static java.util.stream.Collectors.collectingAndThen;
+import static java.util.stream.Collectors.toList;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import kr.momo.exception.MomoException;
+import kr.momo.exception.code.AttendeeErrorCode;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+
+@Getter
+@EqualsAndHashCode
+public class AttendeeGroup {
+
+    private final List<Attendee> attendees;
+
+    public AttendeeGroup(List<Attendee> attendees) {
+        validateUniqueNames(attendees);
+        this.attendees = attendees;
+    }
+
+    private void validateUniqueNames(List<Attendee> attendees) {
+        if (isNotUnique(attendees)) {
+            throw new MomoException(AttendeeErrorCode.DUPLICATED_ATTENDEE_NAME);
+        }
+    }
+
+    private boolean isNotUnique(List<Attendee> attendees) {
+        return !attendees.stream()
+                .map(Attendee::name)
+                .allMatch(new HashSet<>()::add);
+    }
+
+    public AttendeeGroup filterAttendeesByName(List<String> names) {
+        HashSet<String> attendeeNames = new HashSet<>(names);
```

Comment:
추후에 요구사항이 변경될 때 수정에 닫히고 확장에 용이하도록 구현체가 아닌 인터페이스에 의존하는게 좋을 거 같아요!

```suggestion
        Set<String> attendeeNames = new HashSet<>(names);
```